scripts/run_lighthouse_tests_test.py

```diff
@@ -773,3 +773,59 @@ def mock_popen(
                                         '--record_screen',
                                     ]
                                 )
+
+    def test_run_lighthouse_puppeteer_script_creates_directory_when_not_exists(
```

Comment:
We need to check behavior of the function and not implementation. So, instead of mocking os commands, you can create a temporary directory and check if directory is created if it didn't exist.

You can look at `CheckTestsAreCapturedInCiTest` for the example.